dynosaur_derive/src/lib.rs

```diff
@@ -237,15 +262,16 @@ fn mk_dyn_struct_impl_item(struct_ident: &Ident, item_trait: &ItemTrait) -> Toke
              }| {
                  let ident = &sig.ident;
                  let mut sig = sig.clone();
-                 let args = invoke_fn_args(&sig, false);
+                 let (receiver, args) = invoke_fn_args(&sig);
+                 let ref_ = receiver_ref_token(receiver.unwrap());
```

Comment:
This should probably report an error instead of panicking if the function does not have a receiver.

Reply:
Yeah, no idea why did I do that :joy: and btw ... forgot to git add tests.